### PR TITLE
CI: Configure sonarcloud

### PR DESCRIPTION
**Pull request by dweindl**
* Sets up sonarcloud on github actions to run static code analysis for python and C++
* Collect coverage information for sonarcloud
* Moves time-consuming valgrind run to separate workflow
* Cleans up installation scripts
* Fixes a number of minor inconveniences

Progress towards #1053. Still keeping everything on travis ci, while testing whether we are happy with sonarcloud and github actions. To be slimmed down once accepted.

Note that a proper report will only be displayed after sonarscanner has been run on the base branch.

### CODE REVIEW

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5434649&bid=17821477)

File: python/tests/test_misc.py [hunk 1]
```diff
@@ -80,4 +80,9 @@ def test_cmake_compilation(sbml_example_presimulation_module):
 
     cmd = f"set -e; cd {source_dir}; mkdir -p build; cd build; "\
           "cmake ..; make"
-    subprocess.run(cmd, shell=True, capture_output=True, check=True)
+
+    # TODO: when tests run on ubuntu20:
```
Comment (FFroehlich) [inline python/tests/test_misc.py:84]: should be done here or later PR?

Comment (dweindl) [reply python/tests/test_misc.py:84]: Would like to do separately. A number of things still to be improved, but would first like to see the basic version in action.

**Comment by dweindl** [comment]
Where did those memory issues on travis ci come from? :thinking:  

https://travis-ci.com/github/ICB-DCM/AMICI/jobs/325824266
```
[ 86%] Building CXX object CMakeFiles/model_events.dir/wrapfunctions.cpp.o

virtual memory exhausted: Cannot allocate memory

virtual memory exhausted: Cannot allocate memory

cc1plus: out of memory allocating 256 bytes after a total of 2306048 bytes

virtual memory exhausted: Cannot allocate memory

cc1plus: out of memory allocating 2912 bytes after a total of 2347008 bytes

cc1plus: out of memory allocating 32768 bytes after a total of 2441216 bytes

cc1plus: out of memory allocating 2912 bytes after a total of 2306048 bytes

c++: internal compiler error: Killed (program cc1plus)
[...]
```

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5434649&bid=17821477)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5434649&bid=17821477)

**Pull request merged**